### PR TITLE
Add ChannelData checks in TelemetryLoggerMiddleware (#2781)

### LINKED ISSUE #2781: TelemetryLoggerMiddleware doesn't handle missing Teams tenant information
Caused by #2528 

## Versions
- botbuilder v4.10.3
- nodejs version v12.18.1
- Windows Server

## Describe the bug
The [latest version change to the telemetryLoggerMiddleware](https://github.com/microsoft/botbuilder-js/commit/bc95461422f3b3ad74bc0374db7e70e8c6c72a9f#diff-f3bf96f36a847b10436ea6e93f5a1a35R336) introduces an issue where undefined Teams tenant information in the channelData object throws an error on every interaction in Teams-connected bots.
```
onTurnError]: TypeError: Cannot read property 'tenant' of undefined
at TelemetryLoggerMiddleware.populateAdditionalChannelProperties (D:\home\site\wwwroot\node_modules\botbuilder-core\lib\telemetryLoggerMiddleware.js:280:63)
```

## To Reproduce
1. Connect bot to teams tenant that hides its tenent information
2. Trigger a telemetry logging event through a bot connected to teams

## Expected behavior
As the conditional operator is intended to check for channelData.tenant, it should also account for a case in which tenant is undefined.

Changing
`properties['TeamsTenantId'] = channelData.tenant ? channelData.tenant.id : '';`
to
`properties['TeamsTenantId'] = channelData?.tenant ? channelData.tenant.id : '';`
accounts for the tenant property missing and should mitigate this issue.

### PR DESCRIPTION
**Pull request by stevengum**
Fixes #2781

## Specific Changes
- Add truthy checks to TelemetryLoggerMiddleware to avoid throwing an error if Activity.channelData is a falsy value
  - Add test scenario to existing unit test
  - Refactor existing test
- Update `.eslintrc.json` with `*.test.js` overrides
  - Add `env` config for `"mocha"` and `"node"`
  - Ignore `"@typescript-eslint/no-var-requires"` rule for these Node.js JavaScript tests
- Address eslint issues in `telemetryLoggerMiddleware.ts` and `telemetryMiddleware.test.js`.

image

### CODE REVIEW

**Review by johnataylor — APPROVED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -18,6 +18,18 @@
         "libraries/botframework-schema/**/*",
         "libraries/streaming-e2e-tests/**"
     ],
+    "overrides": [
+        {
+            "files": [ "*.test.js" ],
```
Comment (joshgummersall) [inline .eslintrc.json:23]: In the interest in one day writing Typescript tests, perhaps use `{js,ts}` as the extension.

File: libraries/botbuilder-core/src/telemetryLoggerMiddleware.ts [hunk 2]
```diff
@@ -364,15 +364,14 @@ export class TelemetryLoggerMiddleware implements Middleware {
...
             switch (activity.channelId) {
                 case 'msteams':
-                    const channelData = activity.channelData as TeamsChannelData;
-
-                    properties['TeamsTenantId'] = channelData.tenant ? channelData.tenant.id : '';
                     properties['TeamsUserAadObjectId'] = activity.from ? activity.from.aadObjectId : '';
 
-                    if (channelData.team) {
-                        properties['TeamsTeamInfo'] = JSON.stringify(channelData.team);
+                    if (channelData as TeamsChannelData) {
```
Comment (joshgummersall) [inline libraries/botbuilder-core/src/telemetryLoggerMiddleware.ts:372]: I recently added a type guard for this, which you're welcome to use (https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder/src/teamsActivityHelpers.ts#L11).

To me, it's more semantically correct than casting with `as`. This also compiles away to just a truthy check whereas the type guard at least verifies that `channelData` is an object.

The type guard could be expanded to check types of existing properties but I just kept it simple for now.

**Review by joshgummersall — COMMENTED** [review]
Couple quick suggestions.

File: .eslintrc.json (hunk 1 above)
Comment (stevengum) [reply .eslintrc.json:23]: Done!

File: libraries/botbuilder-core/src/telemetryLoggerMiddleware.ts (hunk 2 above)
Comment (joshgummersall) [reply libraries/botbuilder-core/src/telemetryLoggerMiddleware.ts]: Yeah, it's a bit annoying having to write type guards for larger objects. That said, it's a huge dev-exp win both for SDK devs and external devs.

**Pull request merged**